### PR TITLE
CodeCamp #33 [Fix]:add type hints for res_layer, se_layer,normed_predictor,positional_encoding

### PR DESCRIPTION
**Pull request by NoFish-528**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
超级视客营mmdet基础任务，选择链接中任一 part 代码添加 type hints，为part6的res_layer添加type hints

## Modification
为其中的Reslayer和SimplifiedBasicBlock添加了type hint
Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet/models/layers/res_layer.py [hunk 1]
```diff
@@ -114,18 +119,18 @@ class SimplifiedBasicBlock(BaseModule):
...
+                 dilation: int = 1,
+                 downsample: Optional[Sequential] = None,
+                 style: ConfigType = 'pytorch',
+                 with_cp: bool = False,
+                 conv_cfg: OptConfigType = None,
+                 norm_cfg: ConfigType = dict(type='BN'),
+                 dcn: OptConfigType = None,
+                 plugins: OptConfigType = None,
+                 init_fg: OptMultiConfig = None) -> None:
         super(SimplifiedBasicBlock, self).__init__(init_fg)
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/res_layer.py:134]:
```suggestion
        super().__init__(init_fg=init_fg)
```

File: mmdet/models/layers/res_layer.py [hunk 2]
```diff
@@ -168,7 +173,7 @@ def norm2(self):
         """nn.Module: normalization layer after the second convolution layer"""
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/res_layer.py:173]: should also give return type

**Comment by BIGWangYuDong** [comment]
@NoFish-528 Thanks for your PR! You can follow the comments to update your code.

**Comment by NoFish-528** [comment]
> @NoFish-528 Thanks for your PR! You can follow the comments to update your code.

I have updated my code. : ) @BIGWangYuDong

File: mmdet/models/layers/res_layer.py [hunk 3]
```diff
@@ -114,19 +119,19 @@ class SimplifiedBasicBlock(BaseModule):
     expansion = 1
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/res_layer.py:119]: Line 109: super().__init__(*layers)

**Review by BIGWangYuDong — COMMENTED** [review]
Line 109 should update

**Comment by NoFish-528** [comment]
add type hints for res_layer, se_layer,normed_predictor,positional_encoding

File: mmdet/models/layers/normed_predictor.py [hunk 4]
```diff
@@ -50,23 +56,23 @@ class NormedConv2d(nn.Conv2d):
         eps (float, optional): The minimal value of divisor to
              keep numerical stability. Default to 1e-6.
         norm_over_kernel (bool, optional): Normalize over kernel.
-             Default to False.
+             Default: False.
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/normed_predictor.py:59]: Default: -> Defaults to

File: mmdet/models/layers/positional_encoding.py [hunk 5]
```diff
@@ -35,14 +37,14 @@ class SinePositionalEncoding(BaseModule):
...
-                 init_cfg=None):
-        super(SinePositionalEncoding, self).__init__(init_cfg)
+                 num_feats: int,
+                 temperature: int = 10000,
+                 normalize: bool = False,
+                 scale: float = 2 * math.pi,
+                 eps: float = 1e-6,
+                 offset: float = 0.,
+                 init_cfg: OptMultiConfig = None) -> None:
+        super().__init__(init_cfg)
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/positional_encoding.py:47]: init_cfg=init_cfg

File: mmdet/models/layers/positional_encoding.py [hunk 6]
```diff
@@ -54,7 +56,7 @@ def __init__(self,
         self.eps = eps
         self.offset = offset
 
-    def forward(self, mask):
+    def forward(self, mask) -> Tensor:
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/positional_encoding.py:59]: the type of mask

File: mmdet/models/layers/positional_encoding.py [hunk 7]
```diff
@@ -119,19 +121,21 @@ class LearnedPositionalEncoding(BaseModule):
...
-                 init_cfg=dict(type='Uniform', layer='Embedding')):
-        super(LearnedPositionalEncoding, self).__init__(init_cfg)
+    def __init__(
+        self,
+        num_feats: int,
+        row_num_embed: int = 50,
+        col_num_embed: int = 50,
+        init_cfg: MultiConfig = dict(type='Uniform', layer='Embedding')
+    ) -> None:
+        super().__init__(init_cfg)
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/positional_encoding.py:131]: init_cfg=init_cfg

File: mmdet/models/layers/se_layer.py [hunk 8]
```diff
@@ -140,7 +140,7 @@ class ChannelAttention(BaseModule):
     Args:
         channels (int): The input (and output) channels of the attention layer.
         init_cfg (dict or list[dict], optional): Initialization config dict.
-            Defaults to None.
+            Defaults: None
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/se_layer.py:143]: Defaults to None

**Review by BIGWangYuDong — COMMENTED** [review]
1. All `Default:` change to `Defaults to `
2. some typehint are not added
3. add xx=xx in some necessary place, such as super().xxx(xxx=xxx)

**Comment by NoFish-528** [comment]
> 1. All `Default:` change to `Defaults to `
> 2. some typehint are not added
> 3. add xx=xx in some necessary place, such as super().xxx(xxx=xxx)

@BIGWangYuDong Thank you for your suggestion. I update the code with your suggestion

File: mmdet/models/layers/positional_encoding.py [hunk 9]
```diff
@@ -113,25 +115,27 @@ class LearnedPositionalEncoding(BaseModule):
             along x-axis or y-axis. The final returned dimension for
             each position is 2 times of this value.
         row_num_embed (int, optional): The dictionary size of row embeddings.
-            Default 50.
+            Default to 50.
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/positional_encoding.py:118]: Defaults

File: mmdet/models/layers/res_layer.py [hunk 10]
```diff
@@ -12,28 +17,28 @@ class ResLayer(Sequential):
...
         planes (int): planes of block.
         num_blocks (int): number of blocks.
-        stride (int): stride of the first block. Default: 1
+        stride (int): stride of the first block. Default to 1
         avg_down (bool): Use AvgPool instead of stride conv when
-            downsampling in the bottleneck. Default: False
+            downsampling in the bottleneck. Default to False
         conv_cfg (dict): dictionary to construct and config conv layer.
-            Default: None
+            Default to None
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/res_layer.py:24]: Defaults

File: mmdet/models/layers/res_layer.py [hunk 11]
```diff
@@ -114,19 +119,19 @@ class SimplifiedBasicBlock(BaseModule):
...
+                 stride: int = 1,
+                 dilation: int = 1,
+                 downsample: Optional[Sequential] = None,
+                 style: ConfigType = 'pytorch',
+                 with_cp: bool = False,
+                 conv_cfg: OptConfigType = None,
+                 norm_cfg: ConfigType = dict(type='BN'),
+                 dcn: OptConfigType = None,
+                 plugins: OptConfigType = None,
+                 init_fg: OptMultiConfig = None) -> None:
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/res_layer.py:133]: init_fg -> init_cfg

File: mmdet/models/layers/se_layer.py [hunk 12]
```diff
@@ -15,17 +15,17 @@ class SELayer(BaseModule):
     Args:
         channels (int): The input (and output) channels of the SE layer.
         ratio (int): Squeeze ratio in SELayer, the intermediate channel will be
-            ``int(channels/ratio)``. Default: 16.
+            ``int(channels/ratio)``. Default to 16.
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/se_layer.py:18]: Defaults

**Review by BIGWangYuDong — COMMENTED** [review]
There are some questions in this PR, 

1. `Default to` -> `Defaults to` . It should be `Defaults to` in docstrings
2. It is suggested to add docstrings in these python files, I give you some example, which may help you,

**Review by BIGWangYuDong — APPROVED** [review]
LGTM
